onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -1029,6 +1032,34 @@ def _infer_Size(self, node):
             helper.make_tensor_value_info(node.output[0], onnx.TensorProto.INT64, []))
 
     def _infer_Slice(self, node):
+        def less_equal(x, y):
```

Comment:
Would this work? https://docs.sympy.org/latest/modules/core.html?highlight=compare#strictgreaterthan

Reply:
I am not completely sure what `StrictGreaterThan` does but I noticed that whether a relational expression can be determined depends on the ordering and some simple reformulation (I believe what the original code did [here](https://github.com/microsoft/onnxruntime/pull/7401/files#diff-b08185b97bd9eefaa587c3f46213db81d7f57a326a52b542f09339070c0f762bL1082) is similar)

As a simple demonstration, the following
```python
from typing import Type
import sympy
from sympy import StrictLessThan, StrictGreaterThan

x = sympy.Symbol("x", integer=True, positive=True)

def try_bool(e):
    try:
        return bool(e)
    except TypeError:
        return f"Cannot determine {e}"

print(try_bool(sympy.floor(x / 2) < x))
print(try_bool(x > sympy.floor(x / 2)))
print(try_bool(StrictLessThan(sympy.floor(x / 2), x)))
print(try_bool(StrictGreaterThan(x, sympy.floor(x / 2))))
```
prints
```
True
Cannot determine x > floor(x/2)
Cannot determine floor(x/2) < x
Cannot determine x > floor(x/2)
```
Generally, sympy seems to have a higher chance of determining boolean expression when the more complicated expression is on the left hand side.

Reply:
I see, thanks for the detailed experiments. Please add some comments in the code, as it seems like a limitation or bug in sympy. Besides, maybe worth opening an issue in [sympy](https://github.com/sympy/sympy/issues),

Reply:
I see an open issue since 2014. I have added my example as a comment: https://github.com/sympy/sympy/issues/8444